test-suite/src/test/scala/org/scalajs/testsuite/javalib/LongTest.scala

```diff
@@ -114,6 +114,36 @@ object LongTest extends JasmineTest {
       test("-24", -0x24L)
       test("30000", 0x30000L)
       test("-90000", -0x90000L)
+      test("bfc94973", 3217639795L)
+      test("bfc949733", 51482236723L)
+    }
+
+    it("should parse strings in bases 2 to 36") {
+      def test(radix: Int, s: String, v: Long): Unit = {
+        expect(JLong.parseLong(s, radix)).toEqual(v)
+        expect(JLong.valueOf(s, radix).longValue()).toEqual(v)
+      }
+      def testNumber(i: Int): Long = {
+        val result = Long.MaxValue / (if (i == 0) 1 else 1L << i)
+        if (i > 63) -result
+        else result
+      }
+
+      val radices = Array.tabulate[Int](34)(_ + 2)
+      val aNums = Array.tabulate[Long](128)(testNumber(_))
+      for {
+        r <- radices
+        n <- aNums
+      } yield test(r, JLong.toString(n, r), n)
```

Comment:
This loop would be easier to read in this form:
```scala
for {
  radix <- 2 to 36
  i <- 0 until 128
} {
  val n = genTestValue(i)
  test(radix, JLong.toString(n, radix), n)
}
```
Note that there's no `yield`, btw.